src/actions/bookmarks.js

```diff
@@ -0,0 +1,254 @@
+      timeout: API_TIME_OUT,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${jwt}`
+      }
+    }
+    return httpClient.get(url, axiosConfig)
+      .then((res) => {
+        const successAction = buildSuccessActionFromRes(res, types.FETCH_MULTIPLE_BOOKMARKS_SUCCESS)
+        successAction.payload.shouldReplaceExisted = shouldReplaceExisted
+        dispatch(successAction)
+      })
+      .catch((error) => {
+        const failAction = failActionCreators.axios(error, types.FETCH_MULTIPLE_BOOKMARKS_FAILURE)
+        dispatch(failAction)
+      })
+  }
+}
+
+
+/**
+ *
+ *
+ * @export
+ * @param {string} jwt - access_token granted for the user
+ * @param {number} userID - id of user
+ * @param {string} bookmarkSlug - the article slug of the bookmark. Ex: xxx-xxx-xxxxxxx-xx
+ * @param {string} bookmarkHost - the hostname of the bookmark. Ex: 'https://www.xxxx.xx'
+ * @returns
+ */
+export function getSingleBookmark(jwt, userID, bookmarkSlug, bookmarkHost) {
+  return function (dispatch, getState, getHttpClientWithUserCookie) {
+    const url = formAPIURL(`${apiEndpoints.getSingleBookmark(userID, bookmarkSlug)}?host=${bookmarkHost}`, false)
+    const httpClient = getHttpClientWithUserCookie({
+      requestActionType: types.FETCH_SINGLE_BOOKMARK_REQUEST
+    })
+    const axiosConfig = {
+      timeout: API_TIME_OUT,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${jwt}`
+      }
+    }
+    return httpClient.get(url, axiosConfig)
+      .then((res) => {
+        const successAction = buildSuccessActionFromRes(res, types.FETCH_SINGLE_BOOKMARK_REQUEST)
+        dispatch(successAction)
+      })
+      .catch((error) => {
+        const failAction = failActionCreators.axios(error, types.FETCH_SINGLE_BOOKMARK_FAILURE)
+        dispatch(failAction)
+      })
+  }
+}
+
+
+/**
+ *
+ *
+ * @export
+ * @param {string} jwt - access_token granted for the user
+ * @param {number} userID - id of user
+ * @param {number} bookmarkID - id of bookmark
+ * @param {boolean} [byWidget=false] - the delete action is sent from widget or not
+ * @returns
+ */
+export function deleteSingleBookmark(jwt, userID, bookmarkID, byWidget = false) {
+  return function (dispatch, getState, getHttpClientWithUserCookie) {
+    const url = formAPIURL(apiEndpoints.deleteSingleBookmark(userID, bookmarkID), false)
+    const httpClient = getHttpClientWithUserCookie({
+      requestActionType: types.DELETE_SINGLE_BOOKMARK_REQUEST
+    })
+    const axiosConfig = {
+      timeout: API_TIME_OUT,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${jwt}`
+      }
+    }
+    return httpClient.delete(url, axiosConfig)
+      .then((res) => {
+        const successAction = buildSuccessActionFromRes(res, types.DELETE_SINGLE_BOOKMARK_SUCCESS)
+        successAction.payload.bookmarkID = bookmarkID
+        successAction.payload.byWidget = byWidget
```

Comment:
Would the reducer handle `payload` in different ways due to `byWidget`?
Not very sure why we need `byWidget` in our payload.

Reply:
legacy code forgot to remove